snap/info_snap_yaml.go

```diff
@@ -440,16 +444,26 @@ func validateAttr(v interface{}) (interface{}, error) {
 		for k, item := range x {
 			kStr, ok := k.(string)
 			if !ok {
-				return nil, fmt.Errorf("non-string key in attribute map: %v", k)
+				return nil, fmt.Errorf("non-string key in %s map: %v", errorContext, k)
```

Comment:
The errorContext handling seems a bit misleading. This message will end up showing "non-string key in attributes map" even if the attributes value is actually a list or anything else. I suggest reverting to the version without a context parameter, and providing the context from the outside as was already (and still is) being done in the attributes call above.